shared_linux/TestStandCommHost/TestStandCommHost.cxx

```diff
@@ -71,8 +68,32 @@ SerialResult TestStandCommHost::get_position(PositionMsgData *position_out, uint
     res = this->recv_message(MSG_ID_POSITION, sizeof(PositionMsgData), timeout_ms);
     if (res != SERIAL_OK) return res;
 
-    PositionMsgData *data_in = (PositionMsgData *)this->received_message().data;
-    position_out->x_counts = NTOHL(&(data_in->x_counts));
-    position_out->y_counts = NTOHL(&(data_in->y_counts));
+    // Copy message data into output struct
+    memcpy(position_out, this->received_message().data, sizeof(PositionMsgData));
+    // Fixup byte order
+    position_out->x_counts = ntohl(position_out->x_counts);
+    position_out->y_counts = ntohl(position_out->y_counts);
+
+    return SERIAL_OK;
+}
+
+SerialResult TestStandCommHost::get_temp(TempData *temp_out, uint32_t timeout_ms)
+{
+    SerialResult res = this->send_basic_msg(MSG_ID_GET_TEMP);
+    if (res != SERIAL_OK) return res;
+
+    res = this->recv_message(MSG_ID_TEMP, sizeof(TempMsgData), timeout_ms);
+    if (res != SERIAL_OK) return res;
+
+    // Copy message data into output struct
+    TempMsgData msg_data;
+    memcpy(&msg_data, this->received_message().data, sizeof(TempMsgData));
+    // Fixup byte order and re-scale
+    temp_out->temp_ambient = (double)ntohl(msg_data.temp_ambient) / temp_data_scaler;
+    temp_out->temp_motor_x = (double)ntohl(msg_data.temp_motor_x) / temp_data_scaler;
+    temp_out->temp_motor_y = (double)ntohl(msg_data.temp_motor_y) / temp_data_scaler;
+    temp_out->temp_mpmt    = (double)ntohl(msg_data.temp_mpmt) / temp_data_scaler;
+    temp_out->temp_optical = (double)ntohl(msg_data.temp_optical) / temp_data_scaler;
```

Comment:
What is the `1E6` scaler?

Reply:
See my reply up above where you asked about the number of decimal places